src/Core/AndroidNative/maui/src/main/java/com/microsoft/maui/PlatformWrapperView.java

```diff
@@ -0,0 +1,70 @@
+package com.microsoft.maui;
+
+import android.content.Context;
+import android.graphics.Canvas;
+import android.graphics.Rect;
+import android.view.View;
+
+import androidx.annotation.NonNull;
+
+public abstract class PlatformWrapperView extends PlatformContentViewGroup {
+    public PlatformWrapperView(Context context) {
+        super(context);
+        this.viewBounds = new Rect();
+        setClipChildren(false);
+        setWillNotDraw(true);
+    }
+
+    private final Rect viewBounds;
+    private boolean hasShadow;
+
+    /**
+     * Set by C#, determining if we need to call drawShadow()
+     * @param hasShadow
+     */
+    protected final void setHasShadow(boolean hasShadow) {
+        this.hasShadow = hasShadow;
+        postInvalidate();
+    }
+
+    @Override
+    protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
+        if (getChildCount() == 0) {
+            super.onMeasure(widthMeasureSpec, heightMeasureSpec);
+            return;
+        }
+
+        View child = getChildAt(0);
+        viewBounds.set(0, 0, MeasureSpec.getSize(widthMeasureSpec), MeasureSpec.getSize(heightMeasureSpec));
+        child.measure(widthMeasureSpec, heightMeasureSpec);
+        setMeasuredDimension(child.getMeasuredWidth(), child.getMeasuredHeight());
+    }
+
+    @Override
+    protected void dispatchDraw(Canvas canvas) {
+        // Only call into C# if there is a Shadow
+        if (hasShadow) {
+            int viewWidth = viewBounds.width();
+            int viewHeight = viewBounds.height();
+            if (getChildCount() > 0)
+            {
+                View child = getChildAt(0);
```

Comment:
Is it guaranteed here that we do have at lest a child? Could it be empty and result in an NRE?